components/device-mgt/org.wso2.carbon.device.mgt.core/src/main/java/org/wso2/carbon/device/mgt/core/dao/impl/DeviceDAOImpl.java

```diff
@@ -301,4 +301,33 @@ private Connection getConnection() throws DeviceManagementDAOException {
         }
     }
 
+    /**
+     * Get device count of all devices.
+     * @return device count
+     * @throws DeviceManagementDAOException
+     */
+    @Override
+    public int getDeviceCount() throws DeviceManagementDAOException {
+        Connection conn = null;
+        PreparedStatement stmt = null;
+        ResultSet resultSet = null;
+        int deviceCount = 0;
+        try {
+            conn = this.getConnection();
+            String selectDBQueryForType = "SELECT COUNT(DM_DEVICE.ID) FROM DM_DEVICE";
+            stmt = conn.prepareStatement(selectDBQueryForType);
+            resultSet = stmt.executeQuery();
+            while (resultSet.next()) {
+                deviceCount = resultSet.getInt(0);
```

Comment:
Fixed in commit [1]
[1] https://github.com/wso2/carbon-device-mgt/commit/60ee259b2b9ec2c7f929b10c9b383023157cbeb9